projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -252,14 +291,44 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         } else if (visibleColumnIndex === 0 && currentRowEl.previousElementSibling &&
             currentRowEl.previousElementSibling.children[0].tagName.toLowerCase() === 'igx-child-grid-row') {
             const gridElem = currentRowEl.previousElementSibling.querySelector('igx-hierarchical-grid');
```

Comment:
If there are more than one rowIsland defined it would select only the first but it should be the last one.